src/Chapter3.hs

```diff
@@ -970,12 +1051,36 @@ implement the following functions:
 
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
+data Day 
+  = Monday 
+  | Tuesday 
+  | Wednesday 
+  | Thursday 
+  | Friday 
+  | Saturday 
+  | Sunday 
+  deriving (Show, Enum)
+
+isWeekend :: Day -> Bool
+isWeekend Saturday = True
+isWeekend Sunday = True
+isWeekend _ = False
+
+nextDay :: Day -> Day
+nextDay Sunday = Monday
+nextDay d = succ d
+
+daysToParty :: Day -> Int
+daysToParty = go 0
+  where
+    go acc Friday = acc
+    go acc d = go (acc+1) (nextDay d)
```

Comment:
Interesting solution :thinking: Though, I think it can be made a bit more efficient by using the `fromEnum` methods of the `Enum` typeclass